src/github.com/getlantern/lantern-mobile/app/src/main/java/org/getlantern/lantern/fragment/FeedFragment.java

```diff
@@ -31,29 +31,22 @@
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        this.mFeedItems = new ArrayList<FeedItem>();
+        this.adapter = new FeedAdapter(getActivity(), mFeedItems);
 
         Bundle bundle = getArguments();
-        this.feedName = bundle.getString("name");
-        this.mFeedItems = Collections.synchronizedList(new ArrayList<FeedItem>());
+        if (bundle != null) {
+            this.feedName = bundle.getString("name");
```

Comment:
Yeah I'm worried here about the feed name never getting set. Ideally to me this would be an invariant of the class -- i.e. make feedName `final` and check any arguments assigning it for validity.